app/src/main/java/com/example/myapplication/MyAccount.java

```diff
@@ -58,4 +77,64 @@ private void makeEditable(EditText editText) {
             editText.setKeyListener(null);
         }
     }
+
+    public void saveData() {
+        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        try {
+            String BASE_URL = "http://192.168.0.106:8080/";
+            Gson gson = new GsonBuilder()
+                    .setLenient()
+                    .create();
+
+            Retrofit retrofit = new Retrofit.Builder()
+                    .baseUrl(BASE_URL)
+                    .addConverterFactory(ScalarsConverterFactory.create())
+                    .addConverterFactory(GsonConverterFactory.create(gson))
+                    .build();
+            conectWithJava = retrofit.create(ConectWithJava.class);
+            updateNumeNrMasina();
+        } catch (Exception e) {
+            Toast.makeText(getApplicationContext(), e.toString(), Toast.LENGTH_LONG).show();
+        }
+
+        Toast.makeText(this, "Data saved", Toast.LENGTH_SHORT).show();
+    }
+
+    private void updateNumeNrMasina() {
+        HashMap<String, String> updatePlate = new HashMap<>();
+        updatePlate.put("nume", userName.getText().toString());
+        updatePlate.put("nrMasina", plateRegister.getText().toString());
+      //   updatePlate.put("nrNou",newPlateRegister.getText().toString());
+
+        Call<String> call = conectWithJava.getID(updatePlate);
+        String test = String.valueOf(call);
+        updatePlate.remove("nrMasina",plateRegister.getText().toString());
+        updatePlate.put("nrMasina", newPlateRegister.getText().toString());
+        call.enqueue(new Callback<String>() {
+            @Override
+            public void onResponse(Call<String> call, Response<String> response) {
+                Call<String> call2 = conectWithJava.updateUser(test, updatePlate);
```

Comment:
The test is response.body().
Need to put the new value for the updatePlate:
updatePlate.put("nrMasina", newPlateRegister.getText().toString());